discovery-infra/utils.py

```diff
@@ -436,3 +471,15 @@ def get_network_leases(network_name):
 
 def create_ip_address_list(node_count, starting_ip_addr):
     return [str(ipaddress.ip_address(starting_ip_addr) + i) for i in range(node_count)]
+
+def set_hosts_roles_based_on_requested_name(client, cluster_id):
```

Comment:
we have set_hosts_roles in start_discovery. Lets have one function in utils that is doing it.

Reply:
set_hosts_roles in start_discovery is tightly dependent on the teraform node implementation. once the node_controller for teraform nodes is implemented we can move to use this wdyt?